version: "3"

networks:
  amit-infracloud-network:
    driver: bridge
services:
  #Custom containers
  web:
    # build:
    #   context: ./Code/
    #   dockerfile: ./Dockerfile
    #   args:
    #     AMIT_WEB_VERSION: 1.0.1
    image: amitkumar15617/amit_url_shortner_web:1.0.1
    container_name: amit_web
    # expose:
    #     - "5000"
    ports:
      - "5000:5000"
    restart: on-failure
    volumes:
      - ./Code:/src
    depends_on:
      - redis
    networks:
      - amit-infracloud-network
    command: python3 app.py
  #Infra Containers
  redis:
    image: "redis:latest"
    container_name: amit_redis
    sysctls:
      - net.core.somaxconn=511 # TODO needs to revaluate https://github.com/docker-library/redis/issues/191#issuecomment-528693994 <https://github.com/docker-library/redis/issues/191#issuecomment-528693994>
    restart: always
    command: redis-server --requirepass "am!7ku3ar" --appendonly yes
    volumes:
      - redis_database:/data
    ports:
      - "6379:6379"
    healthcheck:
      test: ["CMD", "redis-cli", "ping"]
      interval: 20s
      timeout: 10s
      retries: 5
    networks:
      - amit-infracloud-network

volumes:
  redis_database:
